--- a/src/freedreno/vulkan/tu_util.h	2026-09-12 04:11:58.923249447 +0200
+++ b/src/freedreno/vulkan/tu_util.h	2026-09-15 15:51:01.098917475 +0200
@@ -76,6 +76,7 @@
    TU_DEBUG_FORCE_CONCURRENT_BINNING = BITFIELD64_BIT(36),
    TU_DEBUG_COMPUTE_ROUND_ROBIN      = BITFIELD64_BIT(37),
    TU_DEBUG_GMEM_WARMUP              = BITFIELD64_BIT(38),
+   TU_DEBUG_ZT740_SKIP_CCHE          = BITFIELD64_BIT(39),
 };
 
 struct tu_env {

--- a/src/freedreno/vulkan/tu_util.cc	2026-09-12 04:11:58.919249447 +0200
+++ b/src/freedreno/vulkan/tu_util.cc	2026-09-15 15:51:01.106917475 +0200
@@ -58,6 +58,7 @@
    { "forcecb", TU_DEBUG_FORCE_CONCURRENT_BINNING },
    { "computeroundrobin", TU_DEBUG_COMPUTE_ROUND_ROBIN },
    { "gmem_warmup", TU_DEBUG_GMEM_WARMUP },
+   { "zt740_skip_cche", TU_DEBUG_ZT740_SKIP_CCHE },
    { NULL, 0 }
 };
 

--- a/src/freedreno/vulkan/tu_cmd_buffer.cc	2026-09-14 06:45:09.915649412 +0200
+++ b/src/freedreno/vulkan/tu_cmd_buffer.cc	2026-09-15 15:51:01.114917475 +0200
@@ -403,7 +403,8 @@
       tu_emit_event_write<CHIP>(cmd_buffer, cs, FD_CCU_CLEAN_BLIT_CACHE);
    if (CHIP >= A7XX && (flushes & TU_CMD_FLAG_CCHE_INVALIDATE) &&
        /* Invalidating UCHE seems to also invalidate CCHE */
-       !(flushes & TU_CMD_FLAG_CACHE_INVALIDATE)) {
+       !(flushes & TU_CMD_FLAG_CACHE_INVALIDATE) &&
+       !TU_DEBUG(ZT740_SKIP_CCHE)) {
       /* CP_CCHE_INVALIDATE is just a plain register write underneath, so
        * it needs WFI before it, in order to invalidate at the right point.
        */
